docs/integrations/ml-dist.rst

```diff
@@ -0,0 +1,512 @@
+        else:
+            break
+
+    x_test = np.zeros((len(test_pairs), input_len))
+    y_test = np.zeros((len(test_pairs),))
+
+    for index in range(0, len(test_pairs)):
+        (x, y) = test_pairs[index]
+        x_test[index, 0:input_len] = x
+        y_test[index] = y
+
+The data is prepared in the same way as the training data was prepared, and now we can
+predict the test data and look at the accuracy.
+
+.. code-block:: python
+
+    predictions = mlp_model.predict(x_test)
+    predictions = predictions.flatten()
+
+    _, train_acc = mlp_model.evaluate(x_train, y_train, verbose=0)
+    _, test_acc = mlp_model.evaluate(x_test, y_test, verbose=0)
+
+    print("train_acc: ", train_acc)
+    print("test_acc: ", test_acc)
+
+    [1] train_acc: 0.9827990531921387
+    [2] test_acc:  0.8741965293884277
+
+The accuracy will vary slightly between the different executions. The train accuracy
+is nearly perfect, while the test accuracy is at 87%. These percentages apply when the
+training input is 60 minutes and the prediction output would be for the next 720 minutes
+(12 hours). Depending on how many epochs one uses and the size of the timesteps the results
+vary as well.
+
+Saving A Model To S3
+--------------------
+
+The next step is to save the model to a AWS S3 Bucket so we can access the model from
+potentially another application. Since the data we want to predict will not be normalised
+like the model, we also need to save the MinMaxScaler we created before alongside the
+TensorFlow model.
+
+.. code-block:: python
+
+    import os
+    import boto3
+    from joblib import dump
+
+
+    bucket = "some-s3-bucket-name"
+
+    model_name = "some-model-name" + ".h5"
+    mlp_model.save(model_name)
+
+    scaler_name = "some-scaler-name" + "_scaler.pkl"
+    dump(min_max_scaler, scaler_name)
+
+    client = boto3.client("s3")
+    client.upload_file(model_name, bucket, model_name)
+    client.upload_file(scaler_name, bucket, scaler_name)
+
+    os.remove(model_name)
+    os.remove(scaler_name)
+
+Now we should see our model and scaler in the according S3 bucket and we can continue
+with loading it from possibly a different application.
+
+Loading A Model From S3
+-----------------------
+
+Now that the model is saved in an AWS S3 Bucket, we can simply load it from there
+and use it to predict pump sensor data without having to first train a model.
+
+.. code-block:: python
+
+    bucket = "some-s3-bucket-name"
+
+    model_name = "some-model-name"
+    scaler_name = "some-scaler-name"
+
+    client = boto3.client("s3")
+    client.download_file(bucket, model_name, model_name)
+    client.download_file(bucket, scaler_name, scaler_name)
+
+
+Predicting With Pre-Trained Model
+---------------------------------
+
+The models are now available, this means we can load them from file and start preparing
+the data to predict it. For simplicity we will take the same data that we used to
+train and test the model.
```

Comment:
I think we can leave out the section about using the same data. We can fake that we're getting in new data from somewhere else, it's cooler that way. :wink: